web/api/v1/webkit.go

```diff
@@ -0,0 +1,135 @@
+// Copyright 2020 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1
+
+import (
+	"context"
+	"time"
+
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/pkg/timestamp"
+	"github.com/prometheus/prometheus/storage"
+)
+
+// Webkit is used by the apiv1 and the langServer to retrieve metrics, labels ...etc.
+type webkit struct {
```

Comment:
I think it's depending if you want after this PR another one to refactor the complete API.
If the `webkit` is going to hold the business logic for all endpoint, then maybe another name is needed.

Otherwise, `metadataProvider` is fine for me.

Reply:
Yeah, I don't think we should add more stuff to it, so IMO `metadataProvider` should be fine. Though actually I don't totally like the `Provider` part that much either, as usually I associate `Provider` with switching out the complete source of some data or functionality, rather than just something that provides an intermediate transformation step. But oh well, it's ok for now :)

Reply:
or maybe `metadataService` ?

Reply:
For me that sounds like the same category as `Provider`, but fine with me :)

Reply:
so final conclusion here:
* The prometheus client on langserver side has been renamed to `MetadataService`
* webkit is renamed in favor of `MetatadaProvider`